modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ExpressionTypedValueValueResolver.java

```diff
@@ -42,7 +42,7 @@ public ExpressionTypedValueValueResolver(String expression, Class<Object> expect
     } else {
       DataType expectedDataType =
           DataType.builder().type(expectedClass).mediaType(typedValue.getDataType().getMediaType()).build();
-      T result = (T) transform(typedValue, expectedDataType, event);
+      T result = typedValue.getValue() != null ? (T) transform(typedValue, expectedDataType, event) : null;
```

Comment:
it is weird that you check `typedValue.getValue()` to not be null but then transform the entire `typedValue`. It sounds like this is a check that the transform method should do.